server/platform/services/imageproxy/imageproxy.go

```diff
@@ -87,12 +87,18 @@ func (proxy *ImageProxy) Close() {
 }
 
 func (proxy *ImageProxy) OnConfigChange(oldConfig, newConfig *model.Config) {
-	if *oldConfig.ImageProxySettings.Enable != *newConfig.ImageProxySettings.Enable ||
-		*oldConfig.ImageProxySettings.ImageProxyType != *newConfig.ImageProxySettings.ImageProxyType {
+	if model.AreReferencedValuesEqual(oldConfig.ImageProxySettings.Enable, newConfig.ImageProxySettings.Enable) ||
```

Comment:
Could we just use following?
```go
reflect.DeepEqual(oldConfig.ImageProxySettings, newConfig.ImageProxySettings)
```

Reply:
Yep, I can do that.

For the site URL setting, I'll change that back to how we checked if the settings changed before since it should never be nil in practice. I just realized that the only reason that I needed `AreReferencedValuesEqual` is because I forgot to set defaults on the config I was using for testing